cpp/src/arrow/dataset/scanner_test.cc

```diff
@@ -151,6 +157,64 @@ TEST_F(TestScanner, ToTable) {
   AssertTablesEqual(*expected, *actual);
 }
 
+TEST_F(TestScanner, ToBatches) {
+  SetSchema({field("i32", int32()), field("f64", float64())});
+  auto batch = ConstantArrayGenerator::Zeroes(kBatchSize, schema_);
+
+  for (bool use_threads : {false, true}) {
+    options_->use_threads = use_threads;
+    auto scanner = MakeScanner(batch);
```

Comment:
Is this just testing a scan of one scan task & one batch?  It seems we would want to test more than that.

Reply:
MakeScanner generates a union dataset of 2 InMemoryDatasets each of which repeats the batch 16 times so we should have 32 scan tasks total.